### ISSUE TITLE
Minifier causing errors when code had no errors before.

### ISSUE THREAD

**Issue by Tom-A-Roberts** [malware-dev/MDK-SE#199]
The code didn't initially have errors when compiling in SE. However when I compile in SE after minification, lots of errors appear:

![image](https://user-images.githubusercontent.com/44359621/86628769-44612800-bfc2-11ea-9cb2-38f2c8331c22.png)

These errors seem to be nearly all methods not found. Perhaps it isn't minifying method names where it should?
MDK project here if you wanna take a quick snoop to see if I'm doing something wrong:
https://github.com/ksqk34/Autodocking-2

**Comment by Wicorel** [comment]
I'm guessing it's because it's a static class function.
`
            public static double VectorAngleBetween(Vector3D a, Vector3D b) //returns radians 
            {
                if (a.LengthSquared() == 0 || b.LengthSquared() == 0)
                    return 0;
                else
                    return Math.Acos(MathHelper.Clamp(a.Dot(b) / a.Length() / b.Length(), -1, 1));
            }
`

Comes out as:

`int ò(Vector3D ï, Vector3D î)
    { double ñ = ï.Dot(î); if (ñ < 0) return -1; else return 1; }
    public static double ð(Vector3D ï, Vector3D î)
    {
        if (ï.LengthSquared() == 0 || î.LengthSquared() == 0) return 0;
        else return Math.Acos(MathHelper.Clamp(ï.Dot(î) / ï.Length() / î.
Length(), -1, 1));
    }
`

but reference is not (obviously) using the minified name.

Also, the code generated is not a crazy minified as I'm used to..

**Comment by Wicorel** [comment]
Hmm. the CODE tagging removes some of the white space.. hrmph.

**Comment by ksqk34** [comment]
So you think it could be just an issue with static methods? That's a fairly easy fix if that's the case. I'll just turn them all into standard methods as I can find an instantiated version of the classes in all cases.

**Comment by ksqk34** [comment]
I have tested that, I removed all static methods, turning them into standard ones. This didn't help I'm afraid, it seems to be an issue with standard methods too.

**Comment by Wicorel** [comment]
On reload/rebuild the minified code from your project looks nominal:

(I took out the preserve region for testing)
![image](https://user-images.githubusercontent.com/18159679/86683275-402f0d80-bfb6-11ea-810a-9fa8e8e20b61.png)

It's pasting the code back into VS that auto-formats it..
(I'm using Visual Studio 2017).

It still gets the same error.

My minified code looks like this:
![image](https://user-images.githubusercontent.com/18159679/86682745-bb43f400-bfb5-11ea-8f71-0098358d3fe5.png)

**Comment by malware-dev** [comment]
Interesting.

All I did was run a Resharper Cleanup Code on your project... and it generated correctly.

There is _something_ about the way you write your code that is nonstandard.

**Comment by malware-dev** [comment]
I'm sorry I don't have the energy to figure out the details of how and why this works... but it does. Enclosing the fixed project. Maybe you can figure it out yourself? Or someone else interested?

[Autodocking-2-fixedmaybe.zip](https://github.com/malware-dev/MDK-SE/files/4885423/Autodocking-2-fixedmaybe.zip)

**Comment by ksqk34** [comment]
Looks like I'm downloading Resharper.
I'll see what changes it makes and maybe I'll gain some insight into the crucial differences. I'm glad there is a fix though.

Thanks for your time on this Malware and Wicorel.

**Comment by skiittz** [comment]
I may have a guess as to what is going on.

I started out writing my script using classes nested in partial class Program definitions, as I thought this was necessary to work in the programmable block.

After running into inconsistent accessibility issues and reading another thread, I learned this was not necessary and refactored my code to put all my base classes into the namespace.

But I forgot to remove the partial class declaration from the actual Program.cs.  Resharper caught that this was the only declaration of Program and thus the partial designation was not required and removed it.  

So my best guess is that declaring a class as partial but then not actually having partial implementations is throwing of the brace placement in the minified code.
